### PR TITLE
-- Dynamic Journey iOS Documentation

### PR DESCRIPTION
**Pull request by HafizAhsan1**
Initial documentation for dynamic journey so client's can have an initial impression how workflow will look like. This might be updated again once we complete dynamic journey.

### CODE REVIEW

File: dynamic-journey-ios-guide.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+---
+layout: default
+parent: IDWise Developers
+title: iOS SDK
```
Comment (sakher) [inline dynamic-journey-ios-guide.md:4]: iOS SDK - Advanced Mode

Let's also remove it from the nav and add it as a link to the main iOS SDK file.
Wdyt?

**Comment by sakher** [comment]
@sakher can you please review this and merge if happy to send to Chefz?

File: dynamic-journey-ios-guide.md (hunk 1 above)
Comment (HafizAhsan1) [reply dynamic-journey-ios-guide.md]: Title updated and nav order removed. Link to Example Code is there at the end already.

File: dynamic-journey-ios-guide.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+---
+layout: default
+parent: IDWise Developers
+title: iOS SDK - Advanced Mode
+nav_exclude: true
+---
+
+# IDWise Smart Onboarding SDK for iOS - Advanced Mode
```
Comment (mheskol) [inline dynamic-journey-ios-guide.md:8]: Let's name as Dynamic Mode to unify the terms if this what we agreed to call it for customers and the method

File: dynamic-journey-ios-guide.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+Invoking IDWise SDK is very simple. First import IDWise package in your code file:
+
+```swift
+import IDWise
+```
+
+### Dynamic Journey
+
+In  dynamic journey, IDWise provide the full control to the hosting application to show own UI and handle journey and step events more flexibly. This makes IDWise journey more configurable for the hosting application to not only show desired UI but also to perform any business logic on journey and step events.
```
Comment (mheskol) [inline dynamic-journey-ios-guide.md:67]: Similarly to what I suggested for Android docs if we can rephrase this to become like:
The rest of this document explains how to set up and start using IDWise SDK in Dynamic Journey mode.
Alternatively, if you want to use Simple Journey mode which would get you up and running quicker and with less code if you do not need advanced control over the journey please check Simple Journey Mode here

File: dynamic-journey-ios-guide.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        
+    }
+}
+```
+
+When the journey is started it is assigned a unique id called Journey ID in IDWise system and this is provided as a parameter, `journeyID` with the triggering of `JourneyStarted` event.
+This identifier can be used to fetch the data and status of the journey from IDWise Journey Fetch API at any time.
+
+
+### Starting the Steps
```
Comment (mheskol) [inline dynamic-journey-ios-guide.md:120]: Suggest changing to "Starting Journey Steps"

File: dynamic-journey-ios-guide.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        IDWise.initialize(clientKey: "<YOUR_CLIENT_KEY>") { error in
+                // Deal with error here
+                print(error?.message)
+        }
+        
+        IDWise.startDynamicJourney(journeyDefinitionId: "<YOUR_CUSTOMER_ID>", referenceNumber: "<YOUR_REFERENCE_NO>", locale: "en", journeyDelegate: self, stepDelegate: self)
+        
+``` 
+
+This will make IDWise SDK show a UI with a wizard to guide the user through completing the onboarding journey
```
Comment (mheskol) [inline dynamic-journey-ios-guide.md:85]: I think this is no longer the case as in Dynamic Journey we no longer show our Wizard UI but rather only a camera screen for smart capture
So what we can say:
This will make IDWise SDK start a journey and make it the current journey. You can next start steps of this journey and guide the user through completing the necessary steps in the desired order.

File: dynamic-journey-ios-guide.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+}
+```
+
+When the journey is started it is assigned a unique id called Journey ID in IDWise system and this is provided as a parameter, `journeyID` with the triggering of `JourneyStarted` event.
+This identifier can be used to fetch the data and status of the journey from IDWise Journey Fetch API at any time.
+
+
+### Starting the Steps
+
+When `JourneyStarted` triggered successfuly, you can call the `IDWise.startStep()` to start the specific verfication Step. IDWise.startStep takes the following parameters:
```
Comment (mheskol) [inline dynamic-journey-ios-guide.md:122]: - Suggest we reiterate we called startJourney just to link things together. So maybe we can an initial text to say:
After calling `startJourney` and subsequently when `JourneyStarted` event is triggered successfully ....
- We can either say "you can call the IDWise.startStep method" or say "you can call IDWise.startStep". i.e. if we put "the" article we need to identify the element is a method
- Suggest starting a new line for "IDWise.startStep takes ..... "

File: dynamic-journey-ios-guide.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+```swift
+IDWise.startStep(stepId: Int)
+```
+
+stepId: ID of the step you want to start. (Will be provided by IDWise against each step)
```
Comment (mheskol) [inline dynamic-journey-ios-guide.md:128]: "against" => "for"

File: dynamic-journey-ios-guide.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+Step events will trigger the events in IDWiseSDKStepDelegate passed in startDynamicJourney
```
Comment (mheskol) [inline dynamic-journey-ios-guide.md:129]: This sentence seems a bit disconnected. Initially I thought there is a param called Step events.
Maybe some phrasing like: The step events (provided in stepDelegate parameter provided to startJourney method) will be triggered as step is handled and processed

**Review by mheskol — COMMENTED** [review]
Looks good overall. Left few comments for some minor improvements. I think there is one sentence about the Wizard left from the simple journey docs which is not applicable here.

File: dynamic-journey-ios-guide.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
+---
+layout: default
+parent: IDWise Developers
+title: iOS SDK - Advanced Mode
```
Comment (mheskol) [inline dynamic-journey-ios-guide.md:4]: Let's change every reference of Advanced mode to Dynamic Journey mode to stay consistent and unified with our terms

**Review by mheskol — APPROVED** [review]
